app/src/trackPlayer.ts

```diff
@@ -1,89 +1,100 @@
-    currentTrack: null,
-    currentIndex: null,
-    recordsCount: 0,
-    isPlaying: false
-}
+declare type Handler = (payload: { type: Event; [key: string]: any }) => void;
 
+class TrackPlayer {
+  private static instance: TrackPlayer;
 
+  static getInstance() {
+    if (!TrackPlayer.instance) {
+      TrackPlayer.instance = new TrackPlayer();
+      TrackPlayer.instance.init();
+      return TrackPlayer.instance;
+    }
 
-class TrackPlayer {
-    private static instance: TrackPlayer
+    return TrackPlayer.instance;
+  }
 
-    static getInstance() {
-        if (!TrackPlayer.instance) {
-            TrackPlayer.instance = new TrackPlayer()
-            TrackPlayer.instance.init()
-            return TrackPlayer.instance
-        }
+  private init() {
+    RNTrackPlayer.setupPlayer({});
 
-        return TrackPlayer.instance
-    }
+    const capabilities = [Capability.Play, Capability.Pause, Capability.SeekTo];
+    const options = {
+      stopWithApp: false,
+      capabilities,
+      compactCapabilities: capabilities,
+    };
+    RNTrackPlayer.updateOptions(options);
+    // this.initPlayerQueue();
+  }
 
-    private init() {
-        RNTrackPlayer.setupPlayer({})
-
-        const capabilities = [
-            Capability.Play,
-            Capability.Pause,
-            Capability.SeekTo
-        ]
-        const options = {
-            stopWithApp: false,
-            capabilities,
-            compactCapabilities: capabilities
-        }
-        RNTrackPlayer.updateOptions(options)
-    }
+  public registerService() {
+    console.info("TrackPlayer.registerService()");
+    RNTrackPlayer.registerPlaybackService(() => require("./services/audio"));
+  }
 
-    public registerService() {
-        console.info('TrackPlayer.registerService()')
-        RNTrackPlayer.registerPlaybackService(() => require('./services/audio'))
-    }
+  private async clearQueue() {
+    const queue = await RNTrackPlayer.getQueue();
 
-    private async isInQueue(track: Track): Promise<boolean> {
-        const queue = await RNTrackPlayer.getQueue()
-        const existingTracks = queue.filter((t) => t.id == track.id)
-        if (existingTracks.length > 0) {
-            console.info('Track with id ' + track.id + ' is already in queue.')
-            return true
-        }
-        return false
-    }
+    var indexesToRemove = new Array(queue.length);
+    for (let i = 0; i < queue.length; i++)
+      indexesToRemove[i] = queue.length - 1 - i;
 
-    public async addTrackToQueue(track: Track) {
-        const queue = await RNTrackPlayer.getQueue()
-        const existingTracks = queue.filter((t) => t.id == track.id)
-        const alreadyInqueue = await this.isInQueue(track)
-        if (alreadyInqueue) {
-            return queue
-        }
-        track['isPlaying'] = false
-        RNTrackPlayer.add([track])
-        return RNTrackPlayer.getQueue()
+    for (const id of indexesToRemove) {
+      await RNTrackPlayer.remove([id]);
     }
```

Comment:
nešlo by to celé vyřešit tímto?

```js
await RNTrackPlayer.remove(queue) // queue.reverse() ?
```

Reply:
nene, neslo. Je to issue toho prehravace.